### PR TITLE
Add RealmPrimaryKeyConstraintException.

### LINKED ISSUE #1416: Handling unique (PrimaryKey) constraint violations
Right now Realm throws a generic `io.realm.exceptions.RealmException` with the message "Primary key constraint broken. Value already exists: ..." when you try and insert an object that already exists.

The typical ACID way to handle this (and display an "Already exists" message, etc...) is to catch an exception specific to unique constraint violations.

So unless Realm has another optimum pattern for handling constraint violations I think the exception thrown on constraint violation should be replaced with one more specific.

### PR DESCRIPTION
**Pull request by dalinaum**
Closes #1416.

### CODE REVIEW

**Comment by dalinaum** [comment]
@realm/java review this please :)
refs #1446 too.

File: realm/src/main/java/io/realm/exceptions/RealmPrimaryKeyConstraintException.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.exceptions;
+
+/**
+ * Class for reporting problems with primary key constrain.
```
Comment (cmelchior) [inline realm/src/main/java/io/realm/exceptions/RealmPrimaryKeyConstraintException.java]:
```
Class for reporting problems with the primary key constraint being broken."

@see io.realm.annotations.PrimaryKey
```

**Comment by cmelchior** [comment]
Apart from a small change to the Javadoc for the execption class this looks great :+1:

**Comment by cmelchior** [comment]
Ups. Put the wrong correction in the review message. Updated it.

**Comment by zaki50** [comment]
:+1:

**Comment by dalinaum** [comment]
This PR has conflicts with the current master branch. I should check this again.

**Comment by dalinaum** [comment]
I just resolved. `changelog.txt` was confliced. Please review again @realm/java

**Comment by zaki50** [comment]
:+1:

**Comment by beeender** [comment]
:+1: 1

**Comment by dalinaum** [comment]
Thanks.

**Pull request merged**